### PR TITLE
Feature LANDGRIF-548 calculate impact for new interventions

### PR DESCRIPTION
**Pull request by alexeh**
This PR adds:

Service to calculate Indicator (Impact) Records for new Scenario Interventions.

Updates/Refactors some DTOs and implementation for the Scenario Intervention workflow to match some props and a little performance improvement

Better error handling on GeoCoding, to avoid caching location info that cannot be geocoded and raise and exception

**NOTE:** Tech debt on testing the service including the stored procedures, but you can review this while I am solving the debt

### CODE REVIEW

File: api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts [hunk 1]
```diff
@@ -108,6 +109,8 @@ export class SourcingDataImportService {
           dtoMatchedData.sourcingData,
         );
 
+      // TODO: TBD What to do when there is some location where we cannot determine it's admin-region: i.e coordinates
```
Comment (tiagojsag) [inline api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts:112]:
```suggestion
      // TODO: TBD What to do when there is some location where we cannot determine its admin-region: i.e coordinates
```

File: api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts [hunk 2]
```diff
@@ -116,8 +119,21 @@ export class SourcingDataImportService {
       await this.sourcingLocationService.save(geoCodedSourcingData);
 
       this.logger.log('Generating Indicator Records...');
-      await this.indicatorRecordsService.createIndicatorRecordsForAllSourcingRecords();
-      this.logger.log('Indicator Records generated');
+
+      // TODO: Current approach calculates Impact for all Sourcing Records present in the DB
+      //       Getting H3 data for calculations is done within DB so we need to improve the error handling
+      //       TBD: What to when is no H3 for a Material
```
Comment (tiagojsag) [inline api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts:125]: I can no English derp

**Review by tiagojsag — COMMENTED** [review]
TL;DR

File: api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts (hunk 2 above)
Comment (alexeh) [reply api/src/modules/import-data/sourcing-data/sourcing-data-import.service.ts:125]: LOL I think I had a brain short-circuit
